core-container/entry.sh

```diff
@@ -23,145 +23,133 @@ function echoerr() {
+           sleep 10
         done
     )
+}
+
+function get_kcp_kubeconfig() {
+    wait_kcp_ready
+    kcp_kubeconfig_dir=/home/kubestellar/.kcp
+    kcp_kubeconfig=${kcp_kubeconfig_dir}/admin.kubeconfig
+    echo "Copying the admin.kubeconfig from kubestellar seret..."
+    mkdir -p $kcp_kubeconfig_dir
+    (
+        kubectl --kubeconfig $host_kubeconfig get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > $kcp_kubeconfig
+    )
+}
 
+function create_kcp_provider() {
+    get_kcp_kubeconfig     # kcp is created in a seperate container 
     echo "Delete the kcp provider secret if it already exists, and then create it."
-    if ! kubectl delete secret -n ${NAMESPACE} kcpsec ; then
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete secret -n ${NAMESPACE} kcpsec ; then
         echo "Nothing to delete."
     fi
-    kubectl get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > kcpsecret
-    kubectl create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig="kcpsecret"    
+    kubectl --kubeconfig $host_kubeconfig get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > kcpsecret
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig="kcpsecret"    
     rm kcpsecret
 
     echo "Delete the kcp provider object if it already exists, and then create it."
-    if ! kubectl delete spaceproviderdesc pkcp ; then
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete spaceproviderdesc $PROVIDER_NAME ; then
         echo "Nothing to delete."
     fi
-    kubectl create -f - <<EOF
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkcp
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kcp"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: kcpsec
 EOF
-
-    if [ "$SPACE_PROVIDER" == "kubeflex" ]; then
-        echo "Waiting for the kubeflex provider to be ready... this may take a while."
-        (
-            until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
-                sleep 10
-            done
-        )
+}
  
-        echo "Delete the kubeflex provider object if it already exists, and then create it."
-        if ! kubectl delete spaceproviderdesc pkflex ; then
-            echo "Nothing to delete."
-        fi
-        kubectl create -f - <<EOF
+function create_kubeflex_provider() {
+    echo "Waiting for the kubeflex provider to be ready... this may take a while."
+    (
+        until [ "$(kubectl --kubeconfig $host_kubeconfig get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
```

Comment:
Text processing is fragile. Text searches are susceptible to unintended matches when something else contains the sought string as a substring. Why not do this with `kubectl wait` as demonstrated in https://docs.kubestellar.io/main/Coding%20Milestones/PoC2023q1/environments/dev-env/#create-an-nginx-ingress-controller-with-ssl-passthrough ?

Reply:
True.  I am following the lead here of existing code that checks whether a particular container in the kubestellar pod reached a particular state.  The code that I am following is already running in production.  I agree with what you're saying and the change you are proposing is good and makes sense.  But I am going to open an issue for it and follow up in a future PR.

Reply:
https://github.com/kubestellar/kubestellar/issues/1332